src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -115,12 +119,28 @@ def get_ganesha_conf() -> str:
                 "port": daemon_spec.ports[0] if daemon_spec.ports else 2049,
                 "bind_addr": bind_addr,
                 "haproxy_hosts": [],
+                "nfs_idmap_conf": nfs_idmap_conf,
             }
             if spec.enable_haproxy_protocol:
                 context["haproxy_hosts"] = self._haproxy_hosts()
                 logger.debug("selected haproxy_hosts: %r", context["haproxy_hosts"])
             return self.mgr.template.render('services/nfs/ganesha.conf.j2', context)
 
+        # generate the idmap config
+        def get_idmap_conf() -> str:
+            idmap_conf = spec.idmap_conf
+            cp = ConfigParser()
+            out = StringIO()
+            for sections, keys in idmap_conf:
```

Comment:
mypy seems unhappy with this
```
cephadm/services/nfs.py: note: In function "generate_config":
cephadm/services/nfs.py:134: error: Unpacking a string is disallowed
cephadm/services/nfs.py: note: In function "get_idmap_conf":
cephadm/services/nfs.py:134: error: Unpacking a string is disallowed
cephadm/services/nfs.py: note: In function "generate_config":
cephadm/services/nfs.py:134: error: Item "None" of "Optional[Dict[str, Dict[str, str]]]" has no attribute "__iter__" (not iterable)
cephadm/services/nfs.py: note: In function "get_idmap_conf":
cephadm/services/nfs.py:134: error: Item "None" of "Optional[Dict[str, Dict[str, str]]]" has no attribute "__iter__" (not iterable)
```
I'd guess you need to wrap part of this function in `if idmap_conf is not None`

Reply:
As I replaced the iterations by .read_dict() it should be ok now but I'll check the outcome of mypy